mlir/lib/Target/LLVMIR/Dialect/GENX/GENXToLLVMIRTranslation.cpp

```diff
@@ -25,22 +25,19 @@ using namespace mlir;
 using namespace mlir::LLVM;
 using mlir::LLVM::detail::createIntrinsicCall;
 
-// Create a call to GENX Device function
-// Currently this routine will work only for calling GENX functions that
-// take a single int32 argument.
+// Create a call to SPIR device function.
 static llvm::Value *createDeviceFunctionCall(llvm::IRBuilderBase &builder,
```

Comment:
Generalize to allow passing in the function signature (retType + argument types) and more than one argument.